### PR TITLE
Data connection

### PR DESCRIPTION
**Pull request by oliverhowell**
Quick revision to add to nav and correct some titles...

Not sure we have Data Connections in the right place - WDYT @TomaszGaweda ?

### CODE REVIEW

**Review by Rob-Hazelcast — APPROVED** [review]

File: docs/modules/data-connections/partials/nav.adoc [hunk 1]
```diff
@@ -1,2 +1 @@
-* Data Connections
-** xref:data-connections:data-connections-configuration.adoc[]
+
```
Comment (TomaszGaweda) [inline docs/modules/data-connections/partials/nav.adoc:1]: can we completely delete it now?

File: docs/modules/ROOT/nav.adoc [hunk 2]
```diff
@@ -157,7 +157,11 @@ include::cp-subsystem:partial$nav.adoc[]
 * xref:storage:high-density-memory.adoc[]
 include::tiered-storage:partial$nav.adoc[]
 * xref:cluster-performance:thread-per-core-tpc.adoc[]
-include::data-connections:partial$nav.adoc[]
+* Data Connections
```
Comment (TomaszGaweda) [inline docs/modules/ROOT/nav.adoc:160]: As discussed on Slack, would be best to put in `Develop and Build`

**Review by TomaszGaweda — CHANGES_REQUESTED** [review]

File: docs/modules/data-connections/partials/nav.adoc (hunk 1 above)
Comment (oliverhowell) [reply docs/modules/data-connections/partials/nav.adoc:1]: Yeah, we might as well

File: docs/modules/ROOT/nav.adoc (hunk 2 above)
Comment (oliverhowell) [reply docs/modules/ROOT/nav.adoc]: What do you think of it under the pipelines section as per the latest commit?

Comment (TomaszGaweda) [reply docs/modules/ROOT/nav.adoc]: Perfect 👌

**Review by TomaszGaweda — APPROVED** [review]

**Comment by oliverhowell** [comment]
Should we backport this to 5.5 and other versions? @TomaszGaweda

**Comment by TomaszGaweda** [comment]
Definitely for 5.5 as it's our LTS version. 5.3 and 5.4 is optional IMHO

**Pull request merged**